File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int cp = Integer.MAX_VALUE;
        int prof = 0;
        for(int i = 0;i<prices.length;i++){
            prof = Math.max(prices[i] - cp,prof);
            cp = Math.min(cp,prices[i]);
        }
        return prof;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int cp = INT_MAX;
        int sp = 0;
        for(int i = 0;i<prices.size();i++){
            if(cp > prices[i]) {
                cp = prices[i];
                sp = 0;
            }
            sp = max(sp,prices[i]);
            profit = max(profit,sp-cp);
        }
        return profit;
    }
};